src/coreclr/jit/llvmcodegen.cpp

```diff
@@ -380,10 +381,35 @@ void Llvm::generateUnwindBlocks()
         });
     }
 
+    // The main loop will need to map an outermost mutually protecting region to the innermost. While usually
+    // mutually protecting handlers form a contiguous 'run' in the table, this is not guaranteed, and disjoint
+    // regions, or regions nested inside the respective handlers can 'break up' the run.
```

Comment:
Here's the EH table from the test:
```scala
***************  Exception Handling table
index  eTry, eHnd
  0  ::   6        - Try at BB02..BB02 [000..00C), Handler at BB03..BB03 [00C..015)
  1  ::   4     6  - Try at BB05..BB05 [016..022), Handler at BB06..BB06 [022..02B)
  2  ::   3     4  - Try at BB08..BB08 [02C..038), Handler at BB09..BB09 [038..041)
  3  ::   8     4  - Try at BB08..BB10 [02C..043), Finally at BB11..BB11 [043..04F)
  4  ::   5     6  - Try at BB05..BB05 [016..022), Handler at BB07..BB12 [02B..051)
  5  ::   8     6  - Try at BB05..BB05 [016..022), Handler at BB13..BB13 [051..05A)
  6  ::   7        - Try at BB02..BB02 [000..00C), Handler at BB04..BB14 [015..05C)
  7  ::   8        - Try at BB02..BB02 [000..00C), Handler at BB15..BB15 [05C..065)
  8  ::            - Try at BB02..BB15 [000..065), Finally at BB16..BB16 [065..06C)
```
There are two sets of mutually protecting handlers: {`H0`, `H6`, `H7`} and {`H1`, `H4`, `H5`}.